app/access/api.py

```diff
@@ -42,3 +45,22 @@ def users(request: HttpRequest) -> dict[str, list[UserSchema]]:
     models = User.objects.all()
     responses = [user_to_response(model) for model in models]
     return {"items": responses}
+
+
+@router.delete("users/{username}")
+def delete(request: HttpRequest, username: str) -> HttpResponse:
+    """
+    Delete the user with the given username.
+
+    Return HTTP status code
+    - 204 (No Content) if the user has been deleted
+    - 404 (Not Found) if there is no user with the given username
+    - 500 (Internal Server Error) if there is inconsistency with cognito
+    - 503 (Service Unavailable) if cognito cannot be reached
+    """
+    model = get_object_or_404(User, username=username)
+    deleted = delete_user(model)
+    if not deleted:
+        raise HttpError(500, "Internal Server Error")
+    model.delete()
```

Comment:
I think we should make this whole part atomic and put them in a transaction acquiring a row-lock on the respective object (`select_for_update`). 
This would avoid race conditions and allow to better distinguish different use-cases (e.g. difference between service and cognito vs. user cannot be deleted for other reasons.) This is specially important for logging: currently you use `warning` level when the user cannot be deleted from cognito (https://github.com/geoadmin/service-control/blob/develop/app/cognito/utils/user.py#L36), depeding on the case this is rather `critical` situation in my opinion since it shouldn't happen.

Reply:
Good point! I adjusted it using the transaction (and also changing log level to critical)